### PR TITLE
[improvement](jdbc catalog) when `lower_case_table_names` of jdbc catalog properties is set to `true`, use the real table name to query the jdbc data source

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcClient.java [hunk 1]
```diff
@@ -367,6 +367,20 @@ public List<Column> getColumnsFromJdbc(String dbName, String tableName) {
         return dorisTableSchema;
     }
 
+    public String getRealDatabaseName(String dbname) {
+        if (lowerDBToRealDB.isEmpty()) {
+            getDatabaseNameList();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcClient.java:372]: There may be concurrency issue

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by LemonLiTree — APPROVED** [review]
LGTM

**Pull request merged**